components/private/Math/src/MDCMath.h

```diff
@@ -125,3 +125,35 @@ static inline CGRect MDCRectAlignToScale(CGRect rect, CGFloat scale) {
                     MDCCeil((CGRectGetWidth(rect) + adjustWidthHeight.width) * scale) / scale,
                     MDCCeil((CGRectGetHeight(rect) + adjustWidthHeight.height) * scale) / scale);
 }
+
+static inline CGPoint MDCPointAlignToScale(CGPoint point, CGFloat scale) {
+  if (MDCCGFloatEqual(scale, 0)) {
+    return CGPointZero;
+  }
+
+  return CGPointMake(MDCRound(point.x * scale) / scale, MDCRound(point.y * scale) / scale);
+}
+
+/**
+ Align the centerPoint of a view so that its origin is pixel-aligned. Returns @c CGRectZero if
+ @c scale is zero or @c bounds is @c CGRectNull.
+
+ @param center the unaligned center of the view.
+ @param bounds the bounds of the view.
+ @param scale the native scaling factor for pixel alignment.
+
+ @return the center point of the view such that its origin will be pixel-aligned.
+ */
+static inline CGPoint MDCAlignCenterWithBoundsAndScale(CGPoint center,
+                                                       CGRect bounds,
+                                                       CGFloat scale) {
+  if (MDCCGFloatEqual(scale, 0) || CGRectIsNull(bounds)) {
+    return CGPointZero;
+  }
+
+  CGFloat halfWidth = CGRectGetWidth(bounds) / 2;
```

Comment:
CGRectGetMidX() ?

Reply:
If the bounds are {(5, 10), (15, 20)} then Width is 15 / 2 = 7.55, while mid-X is 12.5.

Reply:
Right, because the bounds may be transformed and the origin may not be 0, 0.
LGTM.